csharp/extractor/Semmle.Extraction.CSharp.Standalone/AssemblyCache.cs

```diff
@@ -165,32 +163,32 @@ public AssemblyInfo ResolveReference(string id)
         /// <returns>The assembly info.</returns>
         public AssemblyInfo GetAssemblyInfo(string filepath)
         {
-            if(assemblyInfo.TryGetValue(filepath, out var info))
+            if (assemblyInfoByFileName.TryGetValue(filepath, out var info))
             {
                 return info;
             }
-            else
+
+            IndexReference(filepath);
```

Comment:
Looks like `filepath` was not added to `failedAssemblyInfoFileNames` before, in case of failure. Good that it is now.